openff/toolkit/topology/topology.py

```diff
@@ -574,9 +574,13 @@ def box_vectors(self, box_vectors):
                 raise InvalidBoxVectorsError(
                     f"Box vectors must be shape (3, 3). Found shape {box_vectors.shape}"
                 )
-        elif isinstance(box_vectors._value, list):
+        elif isinstance(box_vectors, list):
```

Comment:
```suggestion
        # TODO: Add logic to check whether this is a 3x3 list of lists and tell use user to provide a 3x3 numpy array instead
        elif isinstance(box_vectors, list):
```

Reply:
I think it's reasonable a user might try to do that, so I'll add a test for it. It'll probably choke with this code, so I'll also look at refactoring the logic around lists.

Reply:
So ... this clause wasn't actually hit, since `pint.Quantity` converts lists to array-like, resulting it bearing the `shape` attribute. Check out L577 [here](https://app.codecov.io/gh/openforcefield/openff-toolkit/blob/0dbf79be8a8b0768694a88f199d1f815dbe2274d/openff/toolkit/topology/topology.py).

I've added a couple of tests cases to safeguard against other cases, which are now all passing (locally).